File: easy/ts/findMaxArrVal.ts
const arr = [3, 5, 2, 1, 3, 7, 66, 8, 9, 4, 100, 45];

export const findMaxArrVal = (a: number[]): number => {
  let max = a[0];
  for (let i = 1; i < a.length; i++) {
    if (a[i] > max) {
      max = a[i];
    }
  }
  return max;
};
console.log(findMaxArrVal(arr));

// O(n-1)

File: easy/dart/findMaxArrVal.dart
class FindMaxListArr {
  static int result(List<int> arr) {
    int max = arr[0];
    for (int i = 1; i < arr.length; i++) {
      if (arr[i] > max) {
        max = arr[i];
      }
    }
    return max;
  }
}
